swupd/packs.go

```diff
@@ -74,6 +75,44 @@ func (state PackState) String() string {
 	return "invalid"
 }
 
+// CreateAllDeltas builds all of the deltas using the full manifest from one
+// version to the next. This allows better concurrency and the pack creation
+// code can just worry about adding pre-existing files to packs.
+func CreateAllDeltas(outputDir string, fromVersion, toVersion, numWorkers int) error {
+	// Don't try to make deltas for zero packs
+	if fromVersion == 0 {
+		return nil
+	}
+	fromFile := filepath.Join(outputDir, strconv.Itoa(fromVersion), "Manifest.full")
+	toFile := filepath.Join(outputDir, strconv.Itoa(toVersion), "Manifest.full")
+
+	fromManifest, err := ParseManifestFile(fromFile)
+	if err != nil {
+		return err
+	}
+	toManifest, err := ParseManifestFile(toFile)
+	if err != nil {
+		return err
+	}
+
+	if fromVersion >= toVersion {
+		return fmt.Errorf("fromManifest version (%d) must be smaller than toManifest version (%d)", fromVersion, toVersion)
+	}
+
+	var c config
+	c, err = getConfig(filepath.Join(outputDir, ".."))
```

Comment:
Looks like you can just use `getConfig(filepath.Dir(outputDir))` since we aren't creating `outputDir` with a trailing slash... or better safe than sorry.